src/cockpit/transports.py

```diff
@@ -330,31 +328,32 @@ def __init__(self,
                  window: Optional[Dict[str, int]] = None,
                  **kwargs: Any):
         if pty:
-            our_fd, session_fd = os.openpty()
-            kwargs['stderr'] = session_fd
-            self._eio_is_eof = True
-            self._pty_fd = our_fd
+            self._pty_fd, session_fd = os.openpty()
 
             if window is not None:
                 self.set_window_size(**window)
-        else:
-            self._sock, sock = socket.socketpair()
-            our_fd = self._sock.fileno()
-            session_fd = sock.detach()
 
-        try:
+            kwargs['stderr'] = session_fd
             self._process = subprocess.Popen(args,
                                              stdin=session_fd, stdout=session_fd,
                                              start_new_session=True, **kwargs)
-        finally:
             os.close(session_fd)
 
+            in_fd, out_fd = self._pty_fd, self._pty_fd
+            self._eio_is_eof = True
+
+        else:
+            self._process = subprocess.Popen(args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, **kwargs)
```

Comment:
This "Revert "python: use sockets instead of pipes in SubprocessTransport"" commit should drop the test skip for TestStratis.testEncrypted, as it will fix it on c8s.

Reply:
Done, thanks!